packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -763,7 +773,8 @@ class _RawAutocompleteOptionsLayoutDelegate extends SingleChildLayoutDelegate {
       // field of its own. In that case, don't change the constraints width.
       maxWidth: fieldSize.width == 0.0 ? constraints.maxWidth : fieldSize.width,
       maxHeight: max(_kMinUsableHeight, switch (optionsViewOpenDirection) {
-        OptionsViewOpenDirection.down => constraints.maxHeight - fieldOffset.dy - fieldSize.height,
+        OptionsViewOpenDirection.down =>
+          constraints.maxHeight - fieldOffset.dy - fieldSize.height - bottomInset,
```

Comment:
if the calculation "consume"s the bottom inset, then I think typically the bottom inset should be removed from the child widget tree. See `SafeArea.build` for an example.